perf/istio-install/README.md

```diff
@@ -28,53 +28,14 @@ The script will create files to be used later in the setup, as config maps:
 - ${CLUSTER_NAME}/kube.yaml - credentials for accessing k8s
 - ${CLUSTER_NAME}/configmap* - configmaps with GCP-specific configurations
 
-## Setup with operator
+## Setup Istio with different releases
 
-Since Istio 1.4, operator provided by `istioctl` becomes the default installation mechanism.
-`setup_istio_operator.sh` provides the automation. You can add your own operator profile, and then
-setup Istio installation via running the script. For example,
+Below commands will setup istio with corresponding release using [default performance testing overlay file](https://github.com/istio/tools/blob/master/perf/istio-install/istioctl_profiles/default.yaml)
 
-```shell
-export OPERATOR_PROFILE="automtls.yaml" && ./setup_istio_operator.sh
-```
-
-## Setup With Performance Parameters
-
-Look at values.yaml for details on the parameters.
-
-To setup Istio with a specific release, run `DNS_DOMAIN=your-example-domain ./setup_istio.sh release-1.1-20190125-09-16`.
-
-To setup Istio with latest of a branch, run `DNS_DOMAIN=your-example-domain ./setup_istio.sh release-1.2-latest`.
-This command will setup the latest build from the 1.2 release branch.
-
-To just output the deployment file, run `DRY_RUN=1 DNS_DOMAIN=your-example-domain ./setup_istio.sh release-1.1-20190125-09-16`.
-
-### Latest release
-
-DNS_DOMAIN=v11p.qualistio.org ./setup_istio.sh release-1.1-latest
-You may replace the release in the command to the release to test.
+1. To setup Istio with a specific release, run `DNS_DOMAIN=your-example-domain ./setup_istio_release.sh 1.5.0`
 
-You may also override the Helm repo or release URL:
+1. To setup Istio with latest build of a dev stream, run `DNS_DOMAIN=your-example-domain ./setup_istio_release.sh 1.5-dev`.
 
-```bash
-export HELMREPO_URL=https://storage.googleapis.com/istio-release/releases/1.1.0-rc.0/charts/index.yaml
-export RELEASE_URL=https://github.com/istio/istio/releases/download/untagged-c41cff3404b8cc79a97e/istio-1.1.0-rc.0-linux.tar.gz
-
-DNS_DOMAIN=your-example-domain ./setup_istio.sh release-1.1-20190203-09-16
-```
-
-### Overwrite helm flags
-
-To overwrite helm flags, create a file to hold helm flags you want to overwrite and save as extra-values.yaml or other file names.
-
-```bash
-DNS_DOMAIN=your-example-domain EXTRA_VALUES=extra-values.yaml ./setup_istio.sh release-1.1-20190203-09-16
-```
-
-### Installing release candidates
-
-```bash
-DNS_DOMAIN=v11mcp.qualistio.org ./setup_istio_release.sh 1.1.4 pre-release
```

Comment:
setup_istio_releaess.sh also support specific SHA as well right? could you add an example here?

export ISTIO_RELEASE=1.6-alpha.0ef2cd46e2da64b9252c36ca4bf90aa474b73610, something like that.

Reply:
fwiw, I have a script I have started on locally but haven't gotten around to merging, that supports install of any release:

```


WD=$(dirname "$0")
WD=$(cd "$WD"; pwd)
DIRNAME="${WD}/tmp"
mkdir -p "${DIRNAME}"
export GO111MODULE=on

# Passing a tag, like latest or 1.4-dev
if [[ -n "${TAG:-}" ]]; then
  VERSION=$(curl -sL https://gcsweb.istio.io/gcs/istio-build/dev/"${TAG}")
  OUT_FILE="istio-${VERSION}"
  RELEASE_URL="https://storage.googleapis.com/istio-build/dev/${VERSION}/istio-${VERSION}-linux.tar.gz"
# Passing a dev version, like 1.4-alpha.41dee99277dbed4bfb3174dd0448ea941cf117fd
elif [[ -n "${DEV_VERSION:-}" ]]; then
  OUT_FILE="istio-${DEV_VERSION}"
  RELEASE_URL="https://storage.googleapis.com/istio-build/dev/${DEV_VERSION}/istio-${DEV_VERSION}-linux.tar.gz"
# Passing a version, like 1.4.2
elif [[ -n "${VERSION:-}" ]]; then
  OUT_FILE="istio-${VERSION}"
  RELEASE_URL="https://storage.googleapis.com/istio-prerelease/prerelease/${VERSION}/istio-${VERSION}-linux.tar.gz"
# Passing a release url, like https://storage.googleapis.com/istio-prerelease/prerelease/1.4.1/istio-1.4.1-linux.tar.gz
elif [[ -n "${RELEASE_URL:-}" ]]; then
  OUT_FILE=${OUT_FILE:-"$(basename "${RELEASE_URL}" -linux.tar.gz)"}
# Passing a gcs url, like gs://istio-build/dev/1.4-alpha.41dee99277dbed4bfb3174dd0448ea941cf117fd
elif [[ -n "${GCS_URL:-}" ]]; then
  DOWNLOAD_TYPE=gcs
  RELEASE_URL="${GCS_URL}"
  OUT_FILE=${OUT_FILE:-"$(basename "${RELEASE_URL}" -linux.tar.gz)"}
fi

if [[ -z "${RELEASE_URL:-}" ]]; then
  echo "Must set on of TAG, VERSION, DEV_VERSION, RELEASE_URL, GCS_URL"
  exit 2
fi

function download_release() {
  outfile="${DIRNAME}/${OUT_FILE}"
  if [[ ! -d "${outfile}" ]]; then
    tmp=$(mktemp -d)
    if [[ "${DOWNLOAD_TYPE:-}" == gcs ]]; then
      gsutil cp "${GCS_URL}" "${tmp}/out.tar.gz"
      tar xvf "${tmp}/out.tar.gz" -C "${DIRNAME}"
    else
      curl -fJLs -o "${tmp}/out.tar.gz" "${RELEASE_URL}"
      tar xvf "${tmp}/out.tar.gz" -C "${DIRNAME}"
    fi
  else
    echo "${outfile} already exists, skipping download"
  fi
}

function install_istioctl() {
  release=${1:?release folder}
  shift
  "${release}/bin/istioctl" manifest apply --skip-confirmation --wait "${@}"
}

function install_gateways() {
  helm template --set domain="${domain}" base | kubectl -n istio-system apply -f -
}

function install_prom_op() {
  "$WD/setup_prometheus.sh" "${DIRNAME}"
}

download_release
install_istioctl "${DIRNAME}/${OUT_FILE}" "${@}"

install_prom_op
install_gateways
```